pydantic_settings/sources.py

```diff
@@ -23,6 +23,12 @@
 
 DotenvType = Union[Path, List[Path], Tuple[Path, ...]]
 
+# This is used as default value for `_env_file` in `Settings` class and
+# `env_file` in `DotEnvSettingsSource`. Then passing `None` as `_env_file`
+# during initialization of `Settings` has a special meaning and it will
+# override the `env_file` specified in config.
+ENV_FILE_SENTINEL: DotenvType = Path('')
```

Comment:
Can we make this a bit more clear of the exact behavior so I know what will happen without needing to check the logic (right now it says "special meaning", but not what that special meaning is)? Here's an attempt — I haven't checked the logic closely, can you confirm if this is correct?

```suggestion
# This is used as default value for `_env_file` in `Settings` class and
# `env_file` in `DotEnvSettingsSource`. When this value is used, environment
# variables will be read from the `model_config`-specified file. Using this
# as the default instead of `None` allows us to use `None` to mean that environment
# variables should not be read from any env file.
ENV_FILE_SENTINEL: DotenvType = Path('')
```

Another approach we could take is putting this in the `pydantic_settings.main.BaseSettings.__init__` docstring, and then we could shorten/simplify the above comment:
    def __init__(
        __pydantic_self__,
        _case_sensitive: bool | None = None,
        _env_prefix: str | None = None,
        _env_file: DotenvType | None = env_file_sentinel,
        _env_file_encoding: str | None = None,
        _env_nested_delimiter: str | None = None,
        _secrets_dir: str | Path | None = None,
        **values: Any,
    ) -> None:
        """
        TODO: Properly document all additional kwargs here.

        When `_env_file == ENV_FILE_SENTINEL`, the `model_config`-specified env file will be used.
        When `_env_file is None`, environment variables will not be read from any env file.
        """
I do think we need to document all these keyword arguments eventually, but even if you don't want to do that now, I think it's okay to put a placeholder docstring in like this and it would address my concern.

Reply:
Oh I just realized that this is documented in the `pydantic_settings.main.BaseSettings` docstring. I think we can just augment that docstring as part of this PR if you don't mind. I would change the current value:
```
        _env_file: The env file(s) to load settings values from. Defaults to `Path('')`.
```
to
```
        _env_file: The env file(s) to load settings values from. Defaults to `Path('')`, which
        means that the value from `model_config['env_file']` should be used. You can also pass
        `None` to indicate that environment variables should not be loaded from an env file.
```
or similar

Reply:
If we change the `BaseSettings` docstring, I think we can change the comment here to just 
```suggestion
# This is used as default value for `_env_file` in the `BaseSettings` class and
# `env_file` in `DotEnvSettingsSource` so that it can be distinguished from `None`.
# See the docstring of `BaseSettings` for more details.
ENV_FILE_SENTINEL: DotenvType = Path('')
```